src/proxy.js

```diff
@@ -205,9 +195,19 @@ module.exports = function proxyPolyfill() {
     // An alternative here would be to _just_ clone methods to keep behavior consistent.
     let prototypeOk = true;
     if (isMethod || isArray) {
-      if (setProto && proto !== undefined) {
-        setProto(proxy, proto);
-      } else {
+      // Arrays and methods are special: above, we instantiate boring versions of these then swap
+      // our their prototype later. So we only need to use setPrototypeOf in these cases. Some old
+      // engines support `Object.getPrototypeOf` but not `Object.setPrototypeOf`.
+      const setProto =
+        $Object.setPrototypeOf ||
+        ([].__proto__ === Array.prototype
+          ? function setPrototypeOf(O, proto) {
+              validateProto(proto);
+              O.__proto__ = proto;
+              return O;
+            }
+          : noop);
```

Comment:
This will allocate a new `setPrototypeOf` function every time in engines that don't support `Object.setPrototypeOf`, but support `__proto__`, which is why I defined this at the top level alongside `isObject`, `validateProto`, `objectCreate` and `getProto`.

Reply:
You're right, good point. I can revert this.

This came out of me trying to understand (yet again, I wrote most of this code a long time ago) exactly _why_ we needed this, so I can move it up with the comment that we only use it for proxying methods or arrays.